### PR TITLE
Add speculos docs

### PR DESCRIPTION
**Pull request by pscott**
This PR adds a page about speculos. Topics covered are:
- Setup
- Basic usage
- Usage with GDB

The [original speculos repo] is also mentioned twice in the docs, so that users looking for more in-depth explanations know where to look.

### CODE REVIEW

File: source/userspace/speculos.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Usage
+-----
+
+.. warning::
+
+    Only run Speculos on trusted apps, as a malicious app could exploit it and make arbitrary Linux system calls.
+
+To run your application, simply type in::
+
+    ./speculos.py /path/to/app
```
Comment (TamtamHero) [inline source/userspace/speculos.rst:39]: app.elf *

File: source/userspace/speculos.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+You can then use your left / right arrows to emulate the left and right buttons of your device.
+
+Debugging with GDB
+------------------
+
+Speculos can also be used to debug your code with GDB::
+
+    ./speculos.py -d /path/to/app &
+    ./tools/debug.sh /path/to/app
```
Comment (TamtamHero) [inline source/userspace/speculos.rst:49]: app.elf as well

**Review by nndiaye-ledger — APPROVED** [review]

**Review by nndiaye-ledger — COMMENTED** [review]
Hello @pscott that would be great to update this section also https://github.com/LedgerHQ/ledger-dev-doc/blob/master/source/userspace/debugging.rst

**Comment by pscott** [comment]
> Hello @pscott that would be great to update this section also https://github.com/LedgerHQ/ledger-dev-doc/blob/master/source/userspace/debugging.rst

Sure ! What exactly needs an upgrade ? Wording ? Or linking to the speculos section ? Do you have anything special in mind ? :)

**Review by nndiaye-ledger — APPROVED** [review]
Thanks Scott :) ! Perfect

File: source/userspace/speculos.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+To run your application, simply type in::
+
+    ./speculos.py /path/to/app.elf
+
+You can then use your left / right arrows to emulate the left and right buttons of your device.
+
+Communicating with the device
+-----------------------------
+
+You can communicate with the emulated device using `APDUs <https://en.wikipedia.org/wiki/Smart_card_application_protocol_data_unit>`_. Speculos embbeds a TCP server (listening on ``127.0.0.1:999``) to forwards APDUs to the target app.
```
Comment (niooss-ledger) [inline source/userspace/speculos.rst:46]: The port is 9999 by default, not 999

**Pull request merged**